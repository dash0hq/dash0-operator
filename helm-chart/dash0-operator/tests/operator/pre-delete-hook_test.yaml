suite: test pre-delete hook job
templates:
  - operator/pre-delete-hook.yaml
tests:
  - it: pre-delete hook job should match snapshot
    asserts:
      - matchSnapshot: {}

  - it: image tag should default to appVersion
    chart:
      version: 4.5.6
      appVersion: 99.100.101
    asserts:
      - equal:
          path: spec.template.spec.containers[0].image
          value: ghcr.io/dash0hq/operator-controller:99.100.101
