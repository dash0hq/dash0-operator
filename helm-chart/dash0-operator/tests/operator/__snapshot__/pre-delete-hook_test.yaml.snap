pre-delete hook job should match snapshot:
  1: |
    apiVersion: batch/v1
    kind: Job
    metadata:
      annotations:
        helm.sh/hook: pre-delete
        helm.sh/hook-delete-policy: hook-succeeded
      labels:
        app.kubernetes.io/component: pre-delete
        app.kubernetes.io/instance: pre-delete-hook
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: job
        app.kubernetes.io/part-of: dash0-operator
        app.kubernetes.io/version: 1.0.0
        dash0.com/enable: "false"
        helm.sh/chart: dash0-operator-1.0.0
      name: dash0-operator-pre-delete
      namespace: NAMESPACE
    spec:
      backoffLimit: 2
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: pre-delete-hook
            app.kubernetes.io/managed-by: Helm
            helm.sh/chart: dash0-operator-1.0.0
          name: RELEASE-NAME-pre-delete-job
        spec:
          automountServiceAccountToken: true
          containers:
            - command:
                - /manager
                - --uninstrument-all
              image: ghcr.io/dash0hq/operator-controller:1.0.0
              imagePullPolicy: null
              name: pre-delete-job
              resources:
                limits:
                  cpu: 500m
                  ephemeral-storage: 500Mi
                  memory: 128Mi
                requests:
                  cpu: 10m
                  ephemeral-storage: 500Mi
                  memory: 64Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
                runAsNonRoot: true
          restartPolicy: OnFailure
          securityContext:
            runAsNonRoot: true
          serviceAccountName: dash0-operator-controller-manager
